pytest_doctestplus/plugin.py

```diff
@@ -372,26 +372,27 @@ def parse(self, s, name=None):
                         required = re.split(r'\s*[,\s]\s*', match.group(1))
                 elif isinstance(entry, doctest.Example):
 
+                    has_required_modules = DocTestFinderPlus.check_required_modules(required)
+                    if skip_all or skip_next or not has_required_modules:
+                        entry.options[doctest.SKIP] = True
+
+                    elif (config.getoption('remote_data', 'none') != 'any'
+                            and entry.options.get(REMOTE_DATA)):
+                        entry.options[doctest.SKIP] = True
+
                     # If warnings are to be ignored we need to catch them by
                     # wrapping the source in a context manager.
-                    if entry.options.get(IGNORE_WARNINGS, False):
+                    elif entry.options.get(IGNORE_WARNINGS, False):
                         entry.source = ("with _doctestplus_ignore_all_warnings():\n"
                                         + indent(entry.source, '    '))
                         ignore_warnings_context_needed = True
 
                     # Same to show warnings
-                    if entry.options.get(SHOW_WARNINGS, False):
+                    elif entry.options.get(SHOW_WARNINGS, False):
                         entry.source = ("with _doctestplus_show_all_warnings():\n"
                                         + indent(entry.source, '    '))
                         show_warnings_context_needed = True
 
-                    has_required_modules = DocTestFinderPlus.check_required_modules(required)
-                    if skip_all or skip_next or not has_required_modules:
-                        entry.options[doctest.SKIP] = True
-
-                    if config.getoption('remote_data', 'none') != 'any' and entry.options.get(REMOTE_DATA):
-                        entry.options[doctest.SKIP] = True
-
```

Comment:
So the problem was that IGNORE_WARNINGS (and SHOW_WARNINGS) were always checked, so even if the block was skipped a `doctest.Example` with the IGNORE_WARNINGS_CONTEXT was added (below) which would make the full block to be passed instead of skipped.

Reply:
Hmm... What if someone wants to skip warning *inside* a remote data block (maybe out of scope here and to be handled in #137) or the "require" block?

Reply:
That should work, the remote data block is moved above, just to prevent adding the warnings code if the block is already marked as skipped. But if the block is not skipped then the warnings code will be added.